src/App.tsx

```diff
@@ -19,10 +19,31 @@ import useAppStateStore from './store/appStateStore.ts';
 function App() {
     const background = useUIStore((s) => s.background);
     const view = useUIStore((s) => s.view);
+    const setUserMousePosition = useUIStore((s) => s.setUserMousePosition);
+    const setError = useAppStateStore((s) => s.setError);
     const startupInitiated = useRef(false);
     const setSetupDetails = useAppStateStore((s) => s.setSetupDetails);
     const settingUpFinished = useAppStateStore((s) => s.settingUpFinished);
 
+    useEffect(() => {
```

Comment:
perhaps it would be good idea to limit amount of lines we have in App.tsx

You could easily move the lines you've added to a hook named `useUserMousePosition`